source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,142 @@ namespace aspect
+  {
+    // create a quadrature formula based on the compositional element alone.
+    // be defensive about determining that a compositional field actually exists
+    if ( parameters.n_compositional_fields== 0)
+      AssertThrow (introspection.base_elements.compositional_fields
+                   != numbers::invalid_unsigned_int,
+                   ExcMessage("This postprocessor cannot be used without compositional fields."));
+
+    const QGauss<1> quadrature_formula_x(parameters.composition_degree+1);
+    const QGaussLobatto<1> quadrature_formula_y(parameters.composition_degree+1);
+    //FIXME only for 2 dimentional problem now
+    const QAnisotropic<dim> quadrature_formula_xy(quadrature_formula_x,quadrature_formula_y);
+    const QAnisotropic<dim> quadrature_formula_yx(quadrature_formula_y,quadrature_formula_x);
+    const unsigned int n_q_points_xy = quadrature_formula_xy.size();
+    const unsigned int n_q_points_yx = quadrature_formula_yx.size();
+    FEValues<dim> fe_values_xy (mapping,
+                                finite_element,
+                                quadrature_formula_xy,
+                                update_values   |
+                                update_quadrature_points |
+                                update_JxW_values);
+    std::vector<double> compositional_values_xy(n_q_points_xy);
+    FEValues<dim> fe_values_yx (mapping,
+                                finite_element,
+                                quadrature_formula_yx,
+                                update_values   |
+                                update_quadrature_points |
+                                update_JxW_values);
+    std::vector<double> compositional_values_yx(n_q_points_yx);
+
+    std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
+
+    typename DoFHandler<dim>::active_cell_iterator
+    cell = dof_handler.begin_active(),
+    endc = dof_handler.end();
+    double local_area_integrals;
+    std::vector<double> local_compositional_integrals (parameters.n_compositional_fields);
+    std::vector<double> local_compositional_average_integrals (parameters.n_compositional_fields);
+    const double C_max=1.0;//FIXME need to add into the parameters list
+    const double C_min=0;
+    double min_composition=1.0;
+    double max_composition=0;
+
+#define MPI_DG
+
+#ifdef MPI_DG
+    LinearAlgebra::BlockVector distributed_vector (introspection.index_sets.system_partitioning,
+                                                   mpi_communicator);
+    distributed_vector = vector;
+#else
+    LinearAlgebra::BlockVector distributed_vector(vector);
+#endif
+
+    for (; cell != endc; ++cell)
+      {
+        if (cell->is_locally_owned())
+          {
+            cell->get_dof_indices (local_dof_indices);
+            fe_values_xy.reinit (cell);
+            fe_values_yx.reinit (cell);
+
+            local_area_integrals=0;
```

Comment:
you can merge this line with the declaration of local_area_integrals. I would also suggest to rename it to local_area_integral, since it is really only a number. The same applies for the postprocessor (I think).